apis/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -127,7 +127,9 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 	}
 
 	// validate Prometheus config for target allocation
-	if r.Spec.TargetAllocator.Enabled {
+	// we can only validate string-literal configs at admission. ConfigMap configs will be validated at runtime
+	// (when we have access to a client)
+	if r.Spec.TargetAllocator.Enabled && len(r.Spec.ConfigMap.Name) == 0 {
```

Comment:
from what I can tell by the target allocator code, `configToPromConfig` is called at runtime and processes its errors there. So when we write the configmap value to `spec.Config` at the top of the reconcile loop, it should get checked eventually.